### PR TITLE
pkg/api: Fix golint errors

### PR DESCRIPTION
**Pull request by PrasadG193**
**What this PR does / why we need it**:
This PR fixes golint issues in pkg/api

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

**Special notes for your reviewer**:

**Release note**:

```release-note
None
```

### CODE REVIEW

File: pkg/api/testing/compat/compatibility_tester.go [hunk 1]
```diff
@@ -32,7 +32,7 @@ import (
 )
 
 // Based on: https://github.com/openshift/origin/blob/master/pkg/api/compatibility_test.go
-//
+
```
Comment (lavalamp) [inline pkg/api/testing/compat/compatibility_tester.go]: Don't disconnect this section, just move it below the other part of the comment.

**Comment by yue9944882** [comment]
@PrasadG193 thx for fixing this! 

> Fixes #68026 

plz remove this line or it'll close the issue.

File: pkg/api/testapi/testapi.go [hunk 2]
```diff
@@ -322,14 +327,13 @@ func (g TestGroup) SelfLink(resource, name string) string {
...
 		return fmt.Sprintf("/api/%s/%s/%s", g.externalGroupVersion.Version, resource, name)
-	} else {
-		// TODO: will need a /apis prefix once we have proper multi-group
-		// support
-		if name == "" {
-			return fmt.Sprintf("/apis/%s/%s/%s", g.externalGroupVersion.Group, g.externalGroupVersion.Version, resource)
-		}
-		return fmt.Sprintf("/apis/%s/%s/%s/%s", g.externalGroupVersion.Group, g.externalGroupVersion.Version, resource, name)
 	}
+	// TODO: will need a /apis prefix once we have proper multi-group
```
Comment (yue9944882) [inline pkg/api/testapi/testapi.go]: is it necessary for golint?

File: pkg/api/testing/serialization_test.go [hunk 3]
```diff
@@ -65,7 +65,7 @@ func fuzzInternalObject(t *testing.T, forVersion schema.GroupVersion, item runti
 	return item
 }
 
-func Convert_v1beta1_ReplicaSet_to_api_ReplicationController(in *v1beta1.ReplicaSet, out *api.ReplicationController, s conversion.Scope) error {
+func ConvertV1beta1ReplicaSetToAPIReplicationController(in *v1beta1.ReplicaSet, out *api.ReplicationController, s conversion.Scope) error {
```
Comment (yue9944882) [inline pkg/api/testing/serialization_test.go]: this is surprising. does this change make golint happy?

File: pkg/api/v1/pod/util.go [hunk 4]
```diff
@@ -286,20 +286,19 @@ func UpdatePodCondition(status *v1.PodStatus, condition *v1.PodCondition) bool {
 		// We are adding new pod condition.
 		status.Conditions = append(status.Conditions, *condition)
 		return true
-	} else {
-		// We are updating an existing condition, so we need to check if it has changed.
-		if condition.Status == oldCondition.Status {
-			condition.LastTransitionTime = oldCondition.LastTransitionTime
-		}
+	}
```
Comment (yue9944882) [inline pkg/api/v1/pod/util.go]: > is it necessary for golint?

here too. why move this?

File: pkg/api/testing/serialization_test.go (hunk 3 above)
Comment (PrasadG193) [reply pkg/api/testing/serialization_test.go]: Yes. Go doesn't allow underscores in names.

File: pkg/api/testapi/testapi.go (hunk 2 above)
Comment (PrasadG193) [reply pkg/api/testapi/testapi.go]: golint reported this: 
`pkg/api/testapi/testapi.go:325:9: if block ends with a return statement, so drop this else and outdent its block`
Which makes sense, right? If we have return in if itself, then there is no need of else. We can just keep the logic outside of else block.

File: pkg/api/testing/compat/compatibility_tester.go (hunk 1 above)
Comment (PrasadG193) [reply pkg/api/testing/compat/compatibility_tester.go]: Fixed

**Comment by lavalamp** [comment]
please squash, thanks

**Comment by PrasadG193** [comment]
@lavalamp  squashed

**Pull request merged**